Admin/Buckets: re-index & repair + adjust SNS ARN input

nl0 opened the pull request:
### TODO

- [x] wordsmith @akarve

kevinemoore reviewed (approved):
Now that we have a re-index button, we could really use a "refresh" button for the bucket grid (e.g., to see if indexing is finished).

nl0 commented:
@akarve i've updated the UI to use checkbox + text field and it works (tho backend is still failing with 500 when trying to add `quilt-nl0-stage-2` bucket with blank sns arn; seems like this bug is specific to that bucket, since adding other buckets works). i've also adjusted re-index-related copy a little, but pls have another look anyways.
